test/test_environment.py

```diff
@@ -54,8 +54,20 @@ def test_overridden_path(tmp_path):
 
     # mess up PATH, somehow
     with pytest.raises(subprocess.CalledProcessError):
-        utils.cibuildwheel_run(project_dir, output_dir=output_dir, add_env={
-            'CIBW_ENVIRONMENT': '''SOMETHING="$(mkdir new_path && touch new_path/python)" PATH="$(realpath new_path):$PATH"''',
-            'CIBW_ENVIRONMENT_WINDOWS': '''SOMETHING="$(mkdir new_path && type nul > new_path/python.exe)" PATH="$CD\\new_path;$PATH"''',
-        })
+        if utils.platform == 'linux':
+            utils.cibuildwheel_run(project_dir, output_dir=output_dir, add_env={
+                'CIBW_BEFORE_ALL': 'mkdir new_path && touch new_path/python && chmod +x new_path/python',
+                'CIBW_ENVIRONMENT': '''PATH="$(pwd)/new_path:$PATH"''',
+            })
+        else:
+            new_path = tmp_path / 'another_bin'
+            new_path.mkdir()
+            (new_path / 'python').touch(mode=0o777)
```

Comment:
First time I'm seeing that octal literals are useful! :-D

Reply:
First time I'd used them in Python 3, too!